### PR TITLE
4.0.0.beta.9 - WIP

### PR DESCRIPTION
**Pull request by aksonov**
- Enhancement: support inheritance of scene props (useful for `headerMode` attribute for different RN StackNavigator animation)
- Fix: attempt to fix react-navigation pushes instead of jumping when user taps the same tab (#2024)
- Enhancement: use navBarButtonColor or tintColor or headerTintColor for image button tintcolor (#2028)
- Fix: use backBehavior='initialRoute' by default (#2026)
- Enhancement: allow right/leftTitle, right/leftButtonImage to be static functions within scene component
- Enhancement: allow wrapping all screens and navbar buttons with `wrapBy` parameter for `Router`. It could be `observer` from MobX, or `connect` from Redux or any other function.
- Fix: eslint formatting, fix CircleCI build

### CODE REVIEW

**Comment by AlmogRnD** [comment]
Issue with this PR 
screen shot 2017-07-13 at 9 09 09 am

**Comment by aksonov** [comment]
Remove .babelrc

**Comment by AlmogRnD** [comment]
I did that, does nothing same error/issue

File: src/Router.js [hunk 1]
```diff
@@ -160,10 +160,12 @@ function processScene(scene: Scene, inheritProps = {}, clones = []) {
...
   const { tabs, modal, lightbox, navigator, wrap, contentComponent, lazy, drawer, ...parentProps } = scene.props;
 
-  const commonProps = { ...parentProps, ...inheritProps };
+  const commonProps = { ...inheritProps, ...parentProps };
+  delete commonProps.children;
+  delete commonProps.component;
   // add inherit props
   for (const pkey of Object.keys(commonProps)) {
-    if (dontInheritKeys.indexOf(pkey) !== -1) {
+    if (dontInheritKeys.indexOf(pkey) !== -1 && !parentProps[pkey]) {
```
Comment (southerneer) [inline src/Router.js]: more concise/clear version:

`if (dontInheritKeys.includes(pkey) && !parentProps[pkey]) {`

**Comment by aksonov** [comment]
@southerneer Are you able to use that PR?

**Comment by southerneer** [comment]
Yes, this PR works in our project. I did get a babel `decorator-support` error yesterday when pointed to the beta.8 branch that went away when I pointed to my local copy. Looks like you changed that in this PR though. https://github.com/aksonov/react-native-router-flux/pull/2025/files#diff-e56633f72ecc521128b3db6586074d2cL2.

File: src/Router.js (hunk 1 above)
Comment (aksonov) [inline src/Router.js]: Thanks, fixed

**Pull request merged**